From 21314a1c4c47e31e506e93b3cbd87af57c29be47 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Antti=20M=C3=A4=C3=A4tt=C3=A4?= <antti.maatta@qt.io>
Date: Tue, 30 Sep 2025 14:36:54 +0300
Subject: [PATCH 272/553] Add syncqt headers to tracepointgen dependencies

Add the dependencies so that tracepointgen can resolve includes
properly.

Fixes: QTBUG-140443
Change-Id: I57babc653d3d5c975dff3a07f7a6b1fc9323f988
Reviewed-by: Samuli Piippo <samuli.piippo@qt.io>
Reviewed-by: Janne Roine <janne.roine@qt.io>
(cherry picked from commit 02377c3474f02435232ec49d38c691933984a8ad)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 cmake/QtTargetHelpers.cmake | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/cmake/QtTargetHelpers.cmake b/cmake/QtTargetHelpers.cmake
index 37363ed0d26..312138dc858 100644
--- a/cmake/QtTargetHelpers.cmake
+++ b/cmake/QtTargetHelpers.cmake
@@ -1299,9 +1299,12 @@ function(qt_internal_generate_tracepoints name provider)
             set(tracepointgen "${QT_CMAKE_EXPORT_NAMESPACE}::tracepointgen")
         endif()
 
+        add_custom_target(${name}_${provider_name}_sync)
+        qt_internal_add_sync_header_dependencies(${name}_${provider_name}_sync ${name})
+
         add_custom_command(OUTPUT "${tracepoints_path}"
             COMMAND ${tracepointgen} ${provider_name} "${tracepoints_path}"  "I$<JOIN:$<TARGET_PROPERTY:${name},INCLUDE_DIRECTORIES>,;>" ${absolute_file_paths}
-            DEPENDS ${absolute_file_paths}
+            DEPENDS ${absolute_file_paths} ${name}_${provider_name}_sync
             VERBATIM)
         add_custom_target(${name}_${provider_name}_tracepoints_file DEPENDS "${tracepoints_path}")
         add_dependencies(${name} ${name}_${provider_name}_tracepoints_file)
-- 
2.51.2

